### PR TITLE
EXSWHTEC-108 - Implement tests for hipFuncGetAttributes 

### PR DESCRIPTION
**Pull request by mirza-halilcevic**
- Negative parameter tests
- Basic positive test.

Depends-on: #57

### CODE REVIEW

File: catch/unit/executionControl/hipFuncGetAttributes.cc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            GetDeviceAttribute(hipDeviceAttributeMaxSharedMemoryPerBlock, 0));
+  }
+}
+
+TEST_CASE("Unit_hipFuncGetAttributes_Negative_Parameters") {
+  SECTION("attr == nullptr") {
+    HIP_CHECK_ERROR(hipFuncGetAttributes(nullptr, reinterpret_cast<void*>(attribute_test_kernel)),
+                    hipErrorInvalidValue);
+  }
+#if HT_NVIDIA // Disabled on AMD due to defect - EXSWHTEC-241
```
Comment (chrispaquot) [inline catch/unit/executionControl/hipFuncGetAttributes.cc]: Remove this #if and use json for this test case.

Comment (mirza-halilcevic) [reply catch/unit/executionControl/hipFuncGetAttributes.cc]: Done.

**Review by chrispaquot — APPROVED** [review]

**Comment by rorake** [comment]
@mirza-halilcevic 
Will you please resolve the merge conflicts for this PR?

**Comment by rakesroy** [comment]
PR has been merged into develop branch via commit 053c58ab2da663221aaad6265ce2dfdd23cd6695.

**Pull request closed without merging**